arangod/Replication2/ReplicatedLog/LogCommon.h

```diff
@@ -354,16 +354,18 @@ struct CommitFailReason {
   struct QuorumSizeNotReached {
     static auto fromVelocyPack(velocypack::Slice) -> QuorumSizeNotReached;
     void toVelocyPack(velocypack::Builder& builder) const;
+    ParticipantId who;
   };
   struct ForcedParticipantNotInQuorum {
     static auto fromVelocyPack(velocypack::Slice) -> ForcedParticipantNotInQuorum;
     void toVelocyPack(velocypack::Builder& builder) const;
+    ParticipantId who;
   };
   std::variant<NothingToCommit, QuorumSizeNotReached, ForcedParticipantNotInQuorum> value;
 
   static auto withNothingToCommit() noexcept -> CommitFailReason;
-  static auto withQuorumSizeNotReached() noexcept -> CommitFailReason;
-  static auto withForcedParticipantNotInQuorum() noexcept -> CommitFailReason;
+  static auto withQuorumSizeNotReached(ParticipantId const& who) noexcept -> CommitFailReason;
+  static auto withForcedParticipantNotInQuorum(ParticipantId const& who) noexcept -> CommitFailReason;
```

Comment:
I am completely fine with that method, just small thing: Those functions will certainly make a copy of `who`. Making a copy is not `noexcept`. (So this `noexcept` here is actually wrong) You could instead pass the string value. In that case the caller will make the copy for you and you just move your parameter into your member, which is noexcept. 

This has another advantage: if the caller doesn't need that string anymore, it is free to move the string into your function. With that signature above, there will always be a copy, even if the caller could move the string.

The third option would be to provide additional overloads that take a `&&` reference. Then only the overloads that accept `&&` would be noexcept. But this is very often a overkill.

Reply:
ParticipantId is passed by value.